internal/webhook/external/azure.go

```diff
@@ -0,0 +1,219 @@
+package external
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	"github.com/akuity/kargo/internal/git"
+	"github.com/akuity/kargo/internal/helm"
+	xhttp "github.com/akuity/kargo/internal/http"
+	"github.com/akuity/kargo/internal/image"
+	"github.com/akuity/kargo/internal/logging"
+)
+
+const (
+	azureSecretDataKey = "secret"
+	azure              = "azure"
+)
```

Comment:
Tests are going to need to use these values. May as well make them consts.

```suggestion
const (
	azureSecretDataKey = "secret"
	azure              = "azure"

	acrUserAgentPrefix         = "AzureContainerRegistry"
	azureDevOpsUserAgentPrefix = "VSServices"
)
```